### PR TITLE
Use SavedModel format by default in mlflow.keras.log_model

### LINKED ISSUE #3910: [BUG] TF2/Keras autologging failed to log model with default h5 save format
### Willingness to contribute
The MLflow Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the MLflow code base?

- [ ] Yes. I can contribute a fix for this bug independently.
- [x] Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.
- [ ] No. I cannot contribute a bug fix at this time.

### System information
MLflow Version: 1.13.0
TensorFlow version: 2.3.1
Python Version: 3.8.5
Operating System: Mac OS X 10.15.6 19G2021

### Describe the problem

Keras/TF2 models can be saved in the `SavedModel` format from the MLflow version 1.13.0 when `saved_format='tf'` is provided to `mlflow.keras.log_model`. When `mlflow.tensorflow.autolog(log_models=True)` is used, some trained models can't be saved on the train end callback in the default 'h5' format.

It seems that this happens because of https://github.com/mlflow/mlflow/blob/cbe0627428bc2692d851f4180a2ea59073040a63/mlflow/tensorflow.py#L779-L781 This callback calls `mlflow.keras.log_model` with default save format, which is `h5`.

One way to address this is to update `mlflow.tensorflow.autolog` with the `save_format: str = 'h5'` parameter, and propagate `save_format` option to callbacks.

### Code to reproduce issue

```python
import mlflow
import mlflow.tensorflow
import tensorflow as tf
from tensorflow.keras.layers.experimental.preprocessing import TextVectorization

VOCAB_SIZE = 10
SEQUENCE_LENGTH = 16
EMBEDDING_DIM = 32

train_samples = ["this is an example", "another example"]
train_labels = [0.43, 0.23]

mlflow.tensorflow.autolog()

with mlflow.start_run():
    vectorizer_layer = TextVectorization(
        input_shape=(1,),
        max_tokens=VOCAB_SIZE,
        output_mode="int",
        output_sequence_length=SEQUENCE_LENGTH,
    )
    vectorizer_layer.adapt(train_samples)

    model = tf.keras.Sequential(
        [
            vectorizer_layer,
            tf.keras.layers.Embedding(
                VOCAB_SIZE,
                EMBEDDING_DIM,
                name="embedding",
                mask_zero=True,
                input_shape=(1,),
            ),
            tf.keras.layers.GlobalAveragePooling1D(),
            tf.keras.layers.Dense(32, activation="relu"),
            tf.keras.layers.Dense(1, activation="tanh"),
        ]
    )
    model.compile(optimizer="adam", loss="mse", metrics="mae")
    model.fit(train_samples, train_labels, epochs=1)
```

### Other info / logs

```
2020/12/25 16:15:23 INFO mlflow.utils.autologging_utils: tensorflow autologging will track hyperparameters, performance metrics, model artifacts, and lineage information for the current tensorflow workflow to the MLflow run with ID '334a468262d64e7392594a859d2e7da5'
2020-12-25 16:15:23.657399: I tensorflow/core/profiler/lib/profiler_session.cc:164] Profiler session started.
1/1 [==============================] - 0s 31ms/step - loss: 0.1191 - mae: 0.3325
2020/12/25 16:15:24 WARNING mlflow.utils.autologging_utils: MLflow issued a warning during tensorflow autologging: "/Users/user/.pyenv/versions/example/lib/python3.8/site-packages/mlflow/tensorflow.py:786: UserWarning: Logging to MLflow failed: Save or restore weights that is not an instance of `tf.Variable` is not supported in h5, use `save_format='tf'` instead. Got a model or layer TextVectorization with weights [<tensorflow.python.keras.engine.base_layer_utils.TrackableWeightHandler object at 0x14f40a460>]"
```

### What component(s), interfaces, languages, and integrations does this bug affect?

- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

## What changes are proposed in this pull request?

- Use the `SavedModel` format by default in `mlflow.keras.log_model`.
- Fixes #3910

## How is this patch tested?

Existing unit tests

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

`mlflow.sklearn.log_model` now saves a keras model in the `SavedModel` format.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/keras.py [hunk 1]
```diff
@@ -234,9 +234,8 @@ def _is_tf_keras(model):
     with open(os.path.join(data_path, _KERAS_MODULE_SPEC_PATH), "w") as f:
         f.write(keras_module.__name__)
 
-    # By default, Keras uses the SavedModel format -- specified by "tf"
-    # However, we choose to align with prior default of mlflow, HDF5
-    save_format = kwargs.get("save_format", "h5")
+    # Use the SavedModel format if `save_format` is unspecified
+    save_format = kwargs.get("save_format", "tf")
```
Comment (harupy) [inline mlflow/keras.py:238]: Just note that we call `log_model` without `save_format` in the keras autologging:

https://github.com/mlflow/mlflow/blob/ef17afd5766dff33cf2c8d963223eb7be4a61b61/mlflow/keras.py#L679

File: tests/keras/test_keras_model_export.py [hunk 2]
```diff
@@ -204,21 +204,23 @@ def _import_module(name, **kwargs):
         path0 = os.path.join(model_path, "0")
         with pytest.raises(MlflowException):
             mlflow.keras.save_model(x, path0)
-        mlflow.keras.save_model(x, path0, keras_module=FakeKerasModule)
+        mlflow.keras.save_model(x, path0, keras_module=FakeKerasModule, save_format="h5")
```
Comment (harupy) [inline tests/keras/test_keras_model_export.py:207]: Specify `save_format="h5"` in tests that assume a model is saved in the H5 format.

Comment (harupy) [reply tests/keras/test_keras_model_export.py:207]: This implicitly ensures that we can still save a keras model in the H5 format.

**Comment by WeichenXu123** [comment]
* Looks like a behavior change, shall we also update doc ?
* For [BUG] TF2/Keras autologging failed to log model with default h5 save format, does it mean currently some TF model currently does not support H5 format ?

**Review by WeichenXu123 — APPROVED** [review]
LGTM except some.minor comments and pending test pass.

**Comment by harupy** [comment]
> For [BUG] TF2/Keras autologging failed to log model with default h5 save format, does it mean currently some TF model currently does not support H5 format?

Yes :)

**Pull request merged**